### PR TITLE
[Controller] Add ModelH STM32 controller firmware

### PR DESCRIPTION
**Pull request by medoix**
## Description

Looking to merge the ModelH config and keymaps into QMK master. This device is an STM32 controller designed as a drop in replacement for the IBM Model M Keyboard series to support USB Type-B and QMK.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/modelh/chconf.h
Comment (drashna) [inline keyboards/modelh/chconf.h:1]: Please remove.

Defaults have been set in core, and you should only change what is needed to be changed. 

A bunch of the mechlovin boards use the f103, for instance.

File: keyboards/modelh/halconf.h
Comment (drashna) [inline keyboards/modelh/halconf.h:1]: same as chconf.h

File: keyboards/modelh/mcuconf.h
Comment (drashna) [inline keyboards/modelh/mcuconf.h:1]: same as chconf.h

**Review by drashna — CHANGES_REQUESTED** [review]
PR checklist needs to be reviewed. 

Most of the config.h settings are outdated, etc.

File: keyboards/modelh/chconf.h
Comment (medoix) [reply keyboards/modelh/chconf.h:1]: Removed in 90dccdb4eb0759966224d0c90ba02748baa761b6

File: keyboards/modelh/halconf.h
Comment (medoix) [reply keyboards/modelh/halconf.h:1]: Removed in 90dccdb4eb0759966224d0c90ba02748baa761b6

File: keyboards/modelh/mcuconf.h
Comment (medoix) [reply keyboards/modelh/mcuconf.h:1]: Removed in 90dccdb4eb0759966224d0c90ba02748baa761b6

**Comment by drashna** [comment]
identical to #20625

**Comment by sc-steven** [comment]
> identical to #20625

@drashna Yes it appears someone else raised a PR a couple of days after this one. Happy for either to be selected.

**Comment by medoix** [comment]
Closing in favor of this PR https://github.com/qmk/qmk_firmware/pull/20625

**Pull request closed without merging**